servers/src/main/java/tachyon/worker/block/BlockDataManager.java

```diff
@@ -151,13 +151,22 @@ public void addCheckpoint(long sessionId, long fileId) throws TException, IOExce
     String srcPath = PathUtils.concatPath(getSessionUfsTmpFolder(sessionId), fileId);
     String ufsDataFolder =
         mTachyonConf.get(Constants.UNDERFS_DATA_FOLDER, Constants.DEFAULT_DATA_FOLDER);
-    String dstPath = PathUtils.concatPath(ufsDataFolder, fileId);
+    FileInfo fileInfo = mFileSystemMasterClient.getFileInfo(fileId);
+    TachyonURI uri = new TachyonURI(fileInfo.getPath());
+    String dstPath = PathUtils.concatPath(ufsDataFolder, fileInfo.getPath());
```

Comment:
Is your concern code coverage or the missing unit test? There is currently no test class for the BlockDataManager (likely because testing the class requires non-trivial setup), but the code path I changed is covered by tests in the Tachyon Shell module. If your concern is the missing unit test, I am happy to create a proper unit test for BlockDataManager mocking all clients the manager uses.